### PR TITLE
chore(deps): Update client-rekognitionstreaming sdk and custom fetch handler

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes
- Upgrade liveness streaming SDK
- Our CustomWebsocketFetchHandler is mostly copied from the sdk, so update it to match the more recent version but keep custom logic https://github.com/aws/aws-sdk-js-v3/blob/main/packages/middleware-websocket/src/websocket-fetch-handler.ts 

#### Issue #, if available
N/A

#### Description of how you validated changes
tested locally for successful check and connection/regular/session timeouts 

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [x] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/CustomWebSocketFetchHandler.ts [hunk 1]
```diff
@@ -85,10 +85,10 @@ export class CustomWebSocketFetchHandler {
   public readonly metadata: RequestHandlerMetadata = {
     handlerProtocol: 'websocket/h1.1',
   };
-  private readonly configPromise: Promise<WebSocketFetchHandlerOptions>;
+  private config: WebSocketFetchHandlerOptions;
+  private configPromise: Promise<WebSocketFetchHandlerOptions>;
   private readonly httpHandler: RequestHandler<any, any>;
   private readonly sockets: Record<string, WebSocket[]> = {};
-  private readonly utf8decoder = new TextDecoder(); // default 'utf-8' or 'utf8'
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/CustomWebSocketFetchHandler.ts:91]: Was this removed from the sdk?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/CustomWebSocketFetchHandler.ts [hunk 2]
```diff
@@ -218,7 +231,7 @@ export class CustomWebSocketFetchHandler {
       } else {
         resolve({
           done: true,
-          value: undefined as any, // unchecked because done=true.
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/CustomWebSocketFetchHandler.ts:221]: nice!

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/__tests__/CustomWebsocketFetchHandler.test.ts [hunk 3]
```diff
@@ -113,16 +113,44 @@ describe(CustomWebSocketFetchHandler.name, () => {
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         for await (const chunk of responsePayload) {
           /** pass */
-          console.log(chunk);
-          continue;
         }
-      } catch (err) {
+      } catch (err: any) {
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/__tests__/CustomWebsocketFetchHandler.test.ts:117]: Should be `Error` type or `unknown`

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/CustomWebSocketFetchHandler.ts (hunk 1 above)
Comment (esauerbo) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/createStreamingClient/CustomWebSocketFetchHandler.ts:91]: Actually this was custom logic, I'm adding it back rn missed it earlier

**Review by calebpollman — COMMENTED** [review]
Concerned abt the increasing amount of AWS JS SDK code copied in to our codebase between the maintenance burden and the gaps in knowledge since we did not write this code. Is there no other way to approach this?

**Comment by thaddmt** [comment]
> Concerned abt the increasing amount of AWS JS SDK code copied in to our codebase between the maintenance burden and the gaps in knowledge since we did not write this code. Is there no other way to approach this?

It seems like there isn't a good way right now since the SDK is not planning to support returning custom ws closure codes and the rekognition liveness service has a requirement to send custom ws closure codes in order to track error types

**Comment by esauerbo** [comment]
> Concerned abt the increasing amount of AWS JS SDK code copied in to our codebase between the maintenance burden and the gaps in knowledge since we did not write this code. Is there no other way to approach this?

Also our custom logic is applied to private methods which you can't override so I don't think extending the SDK's fetch handler is really an option

**Review by thaddmt — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**